### PR TITLE
refactor: Refactor org_custom_roles.go into multiple files

### PR DESCRIPTION
**Pull request by felixlut**
Continuation of https://github.com/google/go-github/pull/3281#issuecomment-2365259516

The `/organizations/ORGANIZATION_ID/custom_roles` endpoint has been deprecated in favor of `/orgs/ORG/custom-repository-roles`. So the ``orgs_custom_roles.go`` file should be refactored. The file also deals with Organization roles, which is a separate kind of resource to the Organization Repository Roles. For this reason this PR aims to separate the file into two:

* ``orgs_custom_repository_roles.go`` - [API docs](https://docs.github.com/en/enterprise-cloud@latest/rest/orgs/custom-roles?apiVersion=2022-11-28) - `/orgs/ORG/custom-repository-roles`
* `orgs_organization_roles.go` - [API docs](https://docs.github.com/en/rest/orgs/organization-roles?apiVersion=2022-11-28) - `/orgs/ORG/organization-roles`

### CODE REVIEW

File: github/orgs_custom_repository_roles.go
Comment (gmlewis) [inline github/orgs_custom_repository_roles.go:1]:
```suggestion
// Copyright 2024 The go-github AUTHORS. All rights reserved.
```

File: github/orgs_custom_repository_roles_test.go
Comment (gmlewis) [inline github/orgs_custom_repository_roles_test.go:1]:
```suggestion
// Copyright 2024 The go-github AUTHORS. All rights reserved.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @felixlut !
Any time a new file is created, we bump the copyright year to the current year, but if a file is renamed or otherwise edited, we leave the copyright year alone... 
So just a couple minor nits, please, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Review by gmlewis — APPROVED** [review]
Thank you, @felixlut !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by tomfeigin — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @tomfeigin !
Merging.

**Pull request merged**
